internal/controller/model_nodeport.go

```diff
@@ -40,27 +52,61 @@ func NewNodePortLoadBalancerModelGenerator(
 	}
 }
 
-func (g *NodePortLoadBalancerModelGenerator) getDestinationAddresses() (result []string, err error) {
+func isValidIp(ipString string) (err error) {
+	ipParsed := net.ParseIP(ipString)
+	if ipParsed != nil {
+		return nil
+	}
+	return ErrInvalidIpAddress
+}
+
+func isIPv4Address(ipString string) (isIPv4 bool, err error) {
+	return strings.Count(ipString, ":") < 2, isValidIp(ipString)
```

Comment:
Why would a single : be ok?

Reply:
If a Port is defined, but you are completely right. This must not be allowed here.